neqo-transport/src/recovery/mod.rs

```diff
@@ -735,9 +735,14 @@ impl LossRecovery {
 
     /// Calculate when the next timeout is likely to be.  This is the earlier of the loss timer
     /// and the PTO timer; either or both might be disabled, so this can return `None`.
-    pub fn next_timeout(&self, rtt: &RttEstimate) -> Option<Instant> {
+    pub fn next_timeout(&self, path: &Path) -> Option<Instant> {
+        let rtt = path.rtt();
         let loss_time = self.earliest_loss_time(rtt);
-        let pto_time = self.earliest_pto(rtt);
+        let pto_time = if path.amplification_limit() > path.plpmtu() {
```

Comment:
This condition seems like it should be encapsulated better.  In a is_validated() function or something.

Reply:
There is `is_valid()`, but this condition isn't quite the same as `!is_valid()`, since we can also PTO on a path that is not yet validated but where we have sent less than the amplification limit.

How about this?

```suggestion
        let pto_time = if path.is_valid() || path.amplification_limit() > path.plpmtu() {
```

Makes it a bit clearer and we don't need to run through `amplification_limit()` in the common case.

Reply:
That would help, but it might be better to encapsulate that check.

```rust
impl Path {
/// Determine whether we should be setting a PTO for this path.
/// This is true when either the path is valid or when there is enough
/// remaining in the amplification limit to fit a full-sized path (i.e., the path MTU).
pub fn pto_enabled(&self) -> bool {
  // See the implementation `amplification_limit` for details.
  self.amplification_limit() >= self.plpmtu()
}
```